samples/clip-geometry/src/main/java/com/esri/arcgismaps/sample/clipgeometry/components/ClipGeometryViewModel.kt

```diff
@@ -0,0 +1,177 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.clipgeometry.components
+
+import android.app.Application
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.Color
+import com.arcgismaps.geometry.Envelope
+import com.arcgismaps.geometry.Geometry
+import com.arcgismaps.geometry.GeometryEngine
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.symbology.SimpleFillSymbol
+import com.arcgismaps.mapping.symbology.SimpleFillSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.esri.arcgismaps.sample.clipgeometry.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.launch
+
+class ClipGeometryViewModel(application: Application) : AndroidViewModel(application) {
+    // create  a map with a topographic basemap style
+    val arcGISMap = ArcGISMap(BasemapStyle.ArcGISTopographic)
+    // create a MapViewProxy to interact with the MapView
+    val mapViewProxy = MapViewProxy()
+
+    // graphics overlay to display graphics
+    val graphicsOverlay = GraphicsOverlay()
+
+    private var coloradoGraphic  = createColoradoGraphic()
+    private var coloradoFillSymbol = SimpleFillSymbol(
+        SimpleFillSymbolStyle.Solid,
+        Color(R.color.transparentDarkBlue),
+        SimpleLineSymbol(SimpleLineSymbolStyle.Solid, Color.green, 2f)
+    )
+    private var envelopesGraphics = emptyList<Graphic>()
+
+    // Create a message dialog view model for handling error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    val isClipButtonEnabled = mutableStateOf(true)
+    val isResetButtonEnabled = mutableStateOf(false)
+
+    init {
+        viewModelScope.launch {
+            arcGISMap.load().onFailure { error ->
+                messageDialogVM.showMessageDialog(
+                    "Failed to load map",
+                    error.message.toString()
+                )
+            }.onSuccess {
+                // set the viewpoint of the map
+                mapViewProxy.setViewpoint(Viewpoint(40.0, -106.0, 10000000.0))
+
+                coloradoGraphic = createColoradoGraphic()
+                envelopesGraphics = createEnvelopeGraphics()
```

Comment:
Don't think these are needed in the init block, it can just be done during property declaration above